src/store/filters/Sagas.ts

```diff
@@ -1,63 +1,44 @@
 import { takeEvery, all, call, select, put } from "redux-saga/effects";
-import { createNewFilter, FilterDimension } from "../../interfaces/IFilter";
+import { createNewFilter } from "../../interfaces/IFilter";
 import { IProjectInfo } from "../../interfaces/IProjectInfo";
+import { ProjectProperties } from "../../interfaces/IProjectInfo";
+import { ProjectFilterProperties } from "../../interfaces/IProjectInfo";
 import { ProjectsActionTypes } from "../projects/Types";
 import * as projectSelectors from "../projects/Selectors";
 import * as actions from "./Actions";
 
 const DEFAULT_FULL_TEXT_TITLE_VALUE = "";
 
-export function calculateTaskFilters(projects: IProjectInfo[]) {
+function calculateFilterStrings(projects: IProjectInfo[], propertyName : string) {
   const tasks = [
-    ...new Set(projects.flatMap((p) => p.task).filter((x) => x)),
-  ] as string[]; // We know we have no undefined or blank values.
-  return tasks.map((t) => createNewFilter(FilterDimension.Task, t));
-}
-
-export function calculateTypeFilters(projects: IProjectInfo[]) {
-  const types = [
-    ...new Set(projects.flatMap((p) => p.type).filter((x) => x)),
+    ...new Set(projects.flatMap((p) => p.properties[propertyName]).filter((x) => x)),
   ] as string[];
-  return types.map((t) => createNewFilter(FilterDimension.Type, t));
+  return tasks.map((t) => createNewFilter(propertyName, t));
 }
 
-export function calculateLicenseFilters(projects: IProjectInfo[]) {
-  const licenses = [
-    ...new Set(projects.flatMap((p) => p.license).filter((x) => x)),
-  ] as string[];
-  return licenses.map((l) => createNewFilter(FilterDimension.License, l));
-}
-
-export function calculateLanguageFilters(projects: IProjectInfo[]) {
-  const languages = [
-    ...new Set(projects.flatMap((p) => p.language).filter((x) => x)),
-  ] as string[];
-  return languages.map((l) => createNewFilter(FilterDimension.Language, l));
-}
-
-export function calculateTagFilters(projects: IProjectInfo[]) {
-  const tags = [
-    ...new Set(projects.flatMap((p) => p.tags).filter((x) => x)),
-  ] as string[];
-  return tags.map((t) => createNewFilter(FilterDimension.Tags, t));
-}
-
-export function* calculateFilters(): any {
+function* calculateFilters(): any {
   const projects = yield select(projectSelectors.getProjects);
-  const [tasks, types, licenses, languages, tags] = yield all([
-    call(calculateTaskFilters, projects),
-    call(calculateTypeFilters, projects),
-    call(calculateLicenseFilters, projects),
-    call(calculateLanguageFilters, projects),
-    call(calculateTagFilters, projects),
+
+  // TODO_GENERALIZATION: The strings here should be the ProjectFilterProperties keys.
+  // Figure out how to do this "yield-all" thingy to directly create a record....
```

Comment:
I know there's a way to do this but I'm drawing a blank at the moment. I spent about 20 minutes hacking away at this trying to find an alternative for you. I'll keep thinking about it and get back to you if I figure something out.